clients/validator/src/caches/gateways.ts

```diff
@@ -41,4 +42,12 @@ export default class GatewaysCache {
         this.gateways = newGateways
         return newGateways;
     }
+
+    /// Makes requests to assemble a full list of gateways from validator-api
+    async refreshValidatorAPIGateways(url: string): Promise<GatewayBond[]> {
+        const validator_api_url = url.split(":", 2);
+        validator_api_url.push("8080");
```

Comment:
Shouldn't port be part of the URL? What if it's not 8080?

Reply:
It is part of the URL, but it's constructed here instead of passed directly by the user, as is the case with the validator url. Ideally, the user should only pass the host address, and not the port, but some refactoring needs to be done in order to split those. That way, we can attach the necessary port, when needed (1317 for contract calls, 8080 or whatever for validator-api).
As for the port changing, I'll have to look into linking the value from the validator-api crate here. I will however put a global variable in `index.ts`, as it would be easier to maintain.

Reply:
Correct me if I'm wrong, isn't it the case that in theory each validator is going to be able to configure to use a different port hence the user would need to pass one themselves rather than use the hardcoded one?

Reply:
I don't have a problem with asking the user to provide the port, but I'm not sure how the user would come to (reliably) know what port to mention for the validator-api. Having it as a hardcoded value seems like the good old way of doing things -> having one hardcoded port per service e.g. 80 for HTTP, 22 for SSH etc. Of course we can make it configurable on both ends (which at this moment is not, even in the validator-api), but imho it should have a value to default to, in case the user doesn't know what to provide.
If we go with a configurable port here, that change should be made in a future PR, which also makes the validator-api configurable as well.